src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -121,27 +131,93 @@ public String getJavaExpressionWithAppend() {
   }
 
   /**
+   * Holder class for results and result counts from {@link #evaluateColumn}.
+   */
+  public static class EvaluatorResult {
+    private final ReadableColumn results;
+    private final Column<?> resultCounts;
+
+    protected EvaluatorResult(
+        @Nonnull final ReadableColumn results, @Nonnull final Column<?> resultCounts) {
+      this.results = results;
+      this.resultCounts = resultCounts;
+    }
+    /**
+     * @return a {@link ReadableColumn} containing results from {@link #evaluateColumn}
+     */
+    public ReadableColumn getResults() {
```

Comment:
I see what you meant. I agree with you that there is no easy way to avoid this without rethinking how to organize our current tuples/columns classes to provide better abstractions.